components/ai_chat/core/browser/conversation_handler.cc

```diff
@@ -352,6 +346,14 @@ void ConversationHandler::GetState(GetStateCallback callback) {
       current_error_);
 
   std::move(callback).Run(std::move(state));
+
+  /**
+   * When we setup the state of our front-end, we do not initially
+   * have access to details regarding page/context length for the
+   * selected model. Calling `OnModelDataChanged` will cause the
+   * front-end to immediately acquire these details.
+   */
+  OnModelDataChanged();
```

Comment:
Could we pass the data into the callback to avoid a round trip?

i.e. `OnModelDataChanged(NewModelData args)`

Reply:
I originally took this route, but wound up needing to send the page/content length data _also_ through the OnModelDataChanged method. There are a few scenarios: single model use during a session, multi-model use during a session, and modified model use during a session. To cover all bases, making sure Leo always has the right data for the active model, this seemed like the simplest solution.